spdmlib/src/responder/capability_rsp.rs

```diff
@@ -15,17 +18,22 @@ impl ResponderContext {
         bytes: &[u8],
         writer: &'a mut Writer,
     ) -> (SpdmResult, Option<&'a [u8]>) {
-        self.write_spdm_capability_response(bytes, writer);
-
-        (Ok(()), Some(writer.used_slice()))
+        self.write_spdm_capability_response(bytes, writer)
     }
 
-    pub fn write_spdm_capability_response(&mut self, bytes: &[u8], writer: &mut Writer) {
+    pub fn write_spdm_capability_response<'a>(
+        &mut self,
+        bytes: &[u8],
+        writer: &'a mut Writer,
+    ) -> (SpdmResult, Option<&'a [u8]>) {
         if self.common.runtime_info.get_connection_state()
             != SpdmConnectionState::SpdmConnectionAfterVersion
         {
             self.write_spdm_error(SpdmErrorCode::SpdmErrorUnexpectedRequest, 0, writer);
-            return;
+            return (
+                Err(SPDM_STATUS_INVALID_STATE_PEER),
```

Comment:
This is redundant, because the error status is already included in message.

Suggest to change 1st param to processing state. It should be success, if error response is created.

Please check libspdm.

Reply:
> This is redundant, because the error status is already included in message.
> 
> Suggest to change 1st param to processing state. It should be success, if error response is created.
> 
> Please check libspdm.

updated, in latest commit,
the 1st param is removed, kept only the second one,
return None means the message is considered dropped, Some means should send back the response, which includes successful response + error response.

Reply:
updated again, with 1st param added back, but return only Ok(()) in the 1st param.